docs/v3/deploy/static-infrastructure-examples/background-tasks.mdx

```diff
@@ -0,0 +1,454 @@
+
+<Tip>
+Refer to the [examples repository](https://github.com/PrefectHQ/examples/pull/20) for the complete example's source code.
+</Tip>
+
+This pattern is useful when you need to perform operations that are too long for a standard web request-response cycle, such as data processing, sending emails, or interacting with external APIs that might be slow.
+
+
+
+## Overview
+
+This example will build out:
+
+- `@prefect.task` definitions representing the work you want to run in the background
+- A `fastapi` application providing API endpoints to:
+    - Receive task parameters via `POST` request and submit the task to Prefect with `.delay()`
+    - Allow polling for the task's status via a `GET` request using its `task_run_id`
+- A `Dockerfile` to build a multi-stage image for the web app, Prefect server and task worker(s)
+- A `compose.yaml` to manage lifecycles of the web app, Prefect server and task worker(s)
+
+```bash
+├── Dockerfile
+├── README.md
+├── compose.yaml
+├── pyproject.toml
+├── src
+│   └── foo
+│       ├── __init__.py
+│       ├── _internal/*.py
+│       ├── api.py
+│       └── task.py
+```
+
+You can follow along by cloning the [examples repository](https://github.com/PrefectHQ/examples) or instead use [`uv`](https://docs.astral.sh/uv/getting-started/installation/) to bootstrap a your own new project:
+
+```bash
+uv init --lib foo
+uv add prefect marvin
+```
+
+This example application is structured as a library with a `src/foo` directory for portability and organization.
+
+<Note>
+This example does **_not_** require:
+- Prefect Cloud
+- creating a Prefect Deployment
+- creating a work pool
+</Note>
+
+## Useful things to remember
+
+- You can call any Python code from your task definitions (including other flows and tasks!)
+- Prefect [Results](/v3/concepts/caching.html) allow you to save/serialize the `return` value of your task definitions to your result storage (e.g. a local directory, S3, GCS, etc), enabling [caching](/v3/develop/task-caching) and [idempotency](/v3/develop/transactions).
+
+
+## Defining the background task
+
+The core of the background processing is a Python function decorated with `@prefect.task`. This marks the function as a unit of work that Prefect can manage (e.g. observe, cache, retry, etc.)
+
+{/* pmd-metadata: notest */}
+```python src/foo/task.py
+import inspect
+from typing import Any, Callable
+
+import marvin
+
+from prefect import task
+from prefect.cache_policies import INPUTS, TASK_SOURCE
+from prefect.task_worker import serve
+
+
+def def _print_output(output: Any):
+    print(f"result type: {type(output)}")
+    print(f"result: {output!r}")
+
+
+@task(cache_policy=INPUTS + TASK_SOURCE)
+async def cast_data_to_type[T](
+    data: Any,
+    target: type[T],
+    instructions: str,
+    on_complete: Callable[[T], None] | None = _print_output,
+) -> T:
```

Comment:
I love that we're showing off Marvin here, but I wonder if this `cast_data_to_type` is too far in the weeds to highlight what a background task can do.  What if it were something more like having Marvin classify the sentiment of a sentence?  It's still a task worth backgrounding, but it's a little more high level and simpler to grok.

Reply:
yeahhhh that's probably a good point

Reply:
hmm i went to change this, but then had second thoughts. I kind of want to illustrate that the result type can be dynamic, ie that the API layer can do arbitrary logic to infer the result type to cast to

i updated the name of the task to `create_structured_output` - do you think that's still too abstract?

Reply:
Yes I do think it's more clear, and definitely a capability worth showing off.  Still not entirely sure if it should be the first example, but I don't want to slow you down while you're cooking 😜

Reply:
merging for now but will keep this in mind and solicit feedback from users who I know want to use background tasks to see if this is approachable enough